### PR TITLE
Append Sketchfab embed

### PR DESCRIPTION
**Pull request by LavaCxx**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description
Automatic requests from Sentry and GoogleAds are a bit annoying, I'm not very clear on how to block this.

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents
#492 

### Mobile & Desktop Screenshots/Recordings
The default embedded field is very long, actually only part of it was used.
image
image

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/ts/component/block/embed.tsx [hunk 1]
```diff
@@ -645,6 +645,12 @@ const BlockEmbed = observer(class BlockEmbed extends React.Component<I.BlockComp
 						allowScript = !!text.match(/src="https:\/\/gist.github.com([^"]+)"/);
 					};
 
+					if (processor == I.EmbedProcessor.Sketchfab && text.match(/<(iframe|script)/)) {
+						const iframeReg = /<iframe.*?<\/iframe>/;
+						const iframeStr = text.match(iframeReg)?.[0] || '';
+						text = iframeStr;
+					};
```
Comment (ra3orblade) [inline src/ts/component/block/embed.tsx:652]: - Logic conditions should be wrapped in parenthesis `(processor == I.EmbedProcessor.Sketchfab)`
- Make line break after const iframeStr
- I do not perfectly understand how it should work

File: src/ts/lib/util/embed.ts [hunk 2]
```diff
@@ -316,6 +320,7 @@ class UtilEmbed {
 			I.EmbedProcessor.Codepen,
 			I.EmbedProcessor.Bilibili,
 			I.EmbedProcessor.Kroki,
+			I.EmbedProcessor.Sketchfab
```
Comment (ra3orblade) [inline src/ts/lib/util/embed.ts:323]: Add trailing comma

**Review by ra3orblade — CHANGES_REQUESTED** [review]

File: src/ts/component/block/embed.tsx (hunk 1 above)
Comment (LavaCxx) [reply src/ts/component/block/embed.tsx:652]: The original embedded text may contain some remaining HTML structure, which is to extract the iframe inside. After all, not all users will delete the extra parts.
```html
<div class="sketchfab-embed-wrapper"> <iframe title="Steamboat Willie" frameborder="0" allowfullscreen mozallowfullscreen="true" webkitallowfullscreen="true" allow="autoplay; fullscreen; xr-spatial-tracking" xr-spatial-tracking execution-while-out-of-viewport execution-while-not-rendered web-share src="https://sketchfab.com/models/58c5716c49c947f2a5aa152cec39cec0/embed"> </iframe> <p style="font-size: 13px; font-weight: normal; margin: 5px; color: #4A4A4A;"> <a href="https://sketchfab.com/3d-models/steamboat-willie-58c5716c49c947f2a5aa152cec39cec0?utm_medium=embed&utm_campaign=share-popup&utm_content=58c5716c49c947f2a5aa152cec39cec0" target="_blank" rel="nofollow" style="font-weight: bold; color: #1CAAD9;"> Steamboat Willie </a> by <a href="https://sketchfab.com/tramdrey?utm_medium=embed&utm_campaign=share-popup&utm_content=58c5716c49c947f2a5aa152cec39cec0" target="_blank" rel="nofollow" style="font-weight: bold; color: #1CAAD9;"> tramdrey </a> on <a href="https://sketchfab.com?utm_medium=embed&utm_campaign=share-popup&utm_content=58c5716c49c947f2a5aa152cec39cec0" target="_blank" rel="nofollow" style="font-weight: bold; color: #1CAAD9;">Sketchfab</a></p></div>
```

Comment (LavaCxx) [reply src/ts/component/block/embed.tsx:655]: > * Logic conditions should be wrapped in parenthesis `(processor == I.EmbedProcessor.Sketchfab)`
> * Make line break after const iframeStr
> * I do not perfectly understand how it should work

However, I thought again that it seems better to give the initiative to the user.

**Comment by ra3orblade** [comment]
This will be checked and merged after next release as we are currently preparing for it.

**Review by ra3orblade — APPROVED** [review]

**Comment by fuksman** [comment]
@any contributor @LavaCxx code

**Pull request merged**